infra/Pos/Subscription/Dns.hs

```diff
@@ -44,10 +45,15 @@ type KnownRelays = Map NodeId KnownRelay
 activeRelays :: KnownRelays -> [NodeId]
 activeRelays = map fst . filter (relayActive . snd) . M.toList
 
+-- TODO: Use valency and fallbacks
```

Comment:
Yes we should share this logic. Doesn't have to be in this PR (and perhaps best not since testers will need to use these files soon for Windows DNS workarounds).

I hope the situation here is simpler since we only poll the DNS when a subscription fails so hopefully we do not need the cunning STM based "sticky subscriptions" stuff that we need for kademlia.